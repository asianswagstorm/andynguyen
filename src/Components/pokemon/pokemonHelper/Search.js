import React, { Component } from "react";

export default class Search extends Component {
  render() {
    return (
        <form>
          <input onChange = {this.props.onChange} placeholder="Search Pokemon"  type = 'text' name = 'pokename' className="form-control mx-auto" />
        </form>
    );
  }
}
