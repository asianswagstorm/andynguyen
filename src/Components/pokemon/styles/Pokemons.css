
  div.pokemon_row{
    width:100%;
    min-height:70vh;
  }

  div.single_card_1{
    padding:0 20px 20px;
    width:300px
  }

  div.list_pokemon{
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    flex-direction: row;
    align-items: center;
    width: 55%;
    margin-left: auto;
    margin-right: auto; 
  }
  
div.pokemon_search{
    width:400px;
    margin-top:15px;
    margin-left: auto;
    margin-right: auto;
}

  #wrapper {
    min-height:100%;
   }
  
  .form-control{
    color:black;
  }
  
  .no-results{
    width: 300px;
    position: absolute;
    display:inline-block;
    align-self: center;
    right:43%;
    bottom:-300%;
    background-color: red;
  }
  
  .no-results-text{
    text-align: center;
    color: white;
  }
  
  .content{
    padding-top:50px;
    min-height:90vh;
  }
  
   .footer-seperator{
     position:relative;
     background-color: yellow;
     /* width:100%; */
   
  } 