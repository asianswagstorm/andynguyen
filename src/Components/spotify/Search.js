import React, { Component } from "react";
import {filterArtistByName} from "./filterArtist";
import ArtistSuggestions from "./ArtistSuggestions";
import { withRouter } from "../../helpers";
import { connect } from 'react-redux';
const listLimit = 6;

class Search extends Component {
  state = { artistQuery: "", listOfArtist: [], hidden: true, currentSelected: -1}; 

  fetchArtistList = (event) => {
    const artistQuery = event.target.value;
    const listOfArtist =  filterArtistByName(artistQuery);
    this.setState({ listOfArtist, artistQuery,hidden : false, currentSelected: -1 });
  };

  handleKeyPress = event => {
    if (event.key === "Enter") {
      this.searchArtist(this.state.artistQuery);
    }
  };

  selectArtist = (artistQuery, currentSelected) =>  this.setState({ artistQuery, hidden : true ,currentSelected});
  hoverArtist = (artistQuery, currentSelected) =>  this.setState({ artistQuery,  currentSelected});

  searchArtist = (artistQuery) => {
    this.setState({ hidden : true , currentSelected : -1 })
    const {dispatch} = this.props;
    const {getArtist} = this.props.action_props.music_master_action;
    dispatch(getArtist(artistQuery));
  };

  handleKeyDown = (event) => {
      //event.keyCode (down) 40 , (up) 38
     if (this.state.hidden === false && (event.keyCode === 40 || event.keyCode === 38) ) {
          const currentKey = event.keyCode === 40 ? (this.state.currentSelected === listLimit ? 0 : this.state.currentSelected + 1) : 
          (event.keyCode === 38  ? ((this.state.currentSelected <= 0) ? listLimit : this.state.currentSelected - 1) : this.state.currentSelected);
          this.setState({
            currentSelected : currentKey,
            artistQuery : this.state.listOfArtist[currentKey]
          });
    }
  };

  render() {
    const {listOfArtist,artistQuery,hidden,currentSelected} = this.state;
    
    return (
      <div className="artist__search__section">
        <div className="search__artist" id={this.props.noResultsFound.className}>
          <div className = "spotify__input__section">
            <input
                onKeyDown = {this.handleKeyDown}
                list="artists"
                className= "search__input"
                type="search"
                onChange={this.fetchArtistList}
                onKeyPress={this.handleKeyPress}
                onFocus = { () => this.setState({ hidden : true, currentSelected : -1 })}
                placeholder={" Search for an Artist"}
                required 
                value={artistQuery}/> 

            {(listOfArtist.length > 0 &&  hidden === false) && 
                  <ArtistSuggestions  listOfArtist={listOfArtist} 
                                      currentSelected= {currentSelected} 
                                      searchArtist={this.searchArtist} 
                                      hoverArtist= {this.hoverArtist} 
                                      listLimit={listLimit}/>
            }
          </div>
          <button
              className={`search_button ${this.props.noResultsFound.className}`}
              type="submit"
              onClick={() => this.searchArtist(artistQuery)}
            >
              <span className="search__icon"> <i className="fa fa-search"> </i> </span>
          </button>
        </div>

        {this.props.noResultsFound.message !== "" && 
          <div className="invalid__search"><strong>{this.props.noResultsFound.message}</strong></div>
        }
      </div>
    );
  }
}

const mapStateToProps = state => { 
  const musicProps  = state.MusicMasterReducer.defaultMusicStates; 
  const {noResultsFound} = musicProps;

  return {noResultsFound};
};

export default withRouter(connect(mapStateToProps)(Search));
