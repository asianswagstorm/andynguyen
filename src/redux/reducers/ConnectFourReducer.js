/**
 * column x row 
 * Clean this up by using a create object. 
 */
const connectFourBoard = [
    {columnNumber : 0 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 1 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 2 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 3 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 4 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 5 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 6 , rowArrays: ["","","","","",""]}
];

export const defaultConnectFourStates = {
    connectFourBoard,
    rowIndexByColumn : [5,5,5,5,5,5,5],
    computerCurrentColumn : 0, //for testing purposes.
    availableCols: [0,1,2,3,4,5,6] 
};

const DEFAULT_STATES = {defaultConnectFourStates: {...defaultConnectFourStates}};

const connectFourReducer = (state = DEFAULT_STATES, action) => {
    switch(action.type) {
    
        case "SET_CONNECT_FOUR_TABLE":  
            return {
                ...state,
                defaultConnectFourStates: {...state.defaultConnectFourStates, connectFourBoard : action.connectFourBoard}
            };
        case "RESET_CONNECT_FOUR_GAME":
            return {
                ...state,
                defaultConnectFourStates: {...state.defaultConnectFourStates, connectFourBoard : action.connectFourBoard, rowIndexByColumn: [...action.rowIndexByColumn],
                                           computerCurrentColumn: 0  }
            };
        case "SET_ROW_FOR_COLUMN":
            return{
                ...state,
                defaultConnectFourStates: {...state.defaultConnectFourStates, rowIndexByColumn : action.rowIndexByColumn}
            };
        case "UPDATE_COMPUTER_COLUMN": 
            return {
                ...state,
                defaultConnectFourStates: {...state.defaultConnectFourStates, computerCurrentColumn : action.computerCurrentColumn}
            };
        case "UPDATE_AVAILABLE_COLUMNS": 
            return {
                ...state,
                defaultConnectFourStates: {...state.defaultConnectFourStates, availableCols : action.availableCols}
            };
        default:
            return state; //save state
    }
};

export default connectFourReducer 