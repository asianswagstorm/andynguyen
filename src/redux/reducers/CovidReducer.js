export const defaultCovidStates = {
    worldCases  : {population: "7,594,000,000", confirmed: 0, recovered: 0, deaths: 0 },
    canadaCases : {population: "38,844,182", confirmed: 0, recovered: 0, deaths: 0 },
    quebecCases : {population: "8,485,400", confirmed: 0, recovered: 0, deaths: 0 },
    montrealCases : {population: "1,784,694", confirmed: 0, recovered: 0, deaths: 0 },
    apiLoaded : {   World : false,
                    Canada : false,
                    Quebec : false,
                    Montreal : false
                }
}

const DEFAULT_STATES = {defaultCovidStates: {...defaultCovidStates}};
const covidReducer = (state = DEFAULT_STATES, action) => {
    switch(action.type) {
        case 'SET_WORLD_CASES':
            return {
                ...state,
                defaultCovidStates: {...state.defaultCovidStates, worldCases: action.worldCases}
            };
        case 'SET_CANADA_CASES':
            return {
                ...state,
                defaultCovidStates: {...state.defaultCovidStates, canadaCases: action.canadaCases}
            }; 
        case 'SET_QUEBEC_CASES':
            return {
                ...state,
                defaultCovidStates: {...state.defaultCovidStates, quebecCases: action.quebecCases}
            };
        case 'SET_MONTREAL_CASES':
            return {
                ...state,
                defaultCovidStates: {...state.defaultCovidStates, montrealCases: action.montrealCases }
            };
        case 'MODIFY_CANADIAN_CASES':
            return {
                ...state,
                defaultCovidStates: {...state.defaultCovidStates, canadaCases: action.canadaCases }
            };
        case 'SET_API_LOADED':
            return {
                ...state, 
                defaultCovidStates: {...state.defaultCovidStates, apiLoaded: action.apiLoaded }
            };
        default:
            return state; 
    } 
};

export default covidReducer 