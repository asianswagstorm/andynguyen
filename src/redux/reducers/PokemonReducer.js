const defaultPokemonStates = {
    pokemons : [],
    hasErrors : false,
    tracker : 0,
    searchedPokemon : "",
    pokemon:{},
    pokemonIndex: 0,
    speciesData:{}
};

const DEFAULT_STATES = {defaultPokemonStates: {...defaultPokemonStates}};

const PokemonReducer = (state = DEFAULT_STATES, action) => {
    switch(action.type) {
        case "SET_POKEMON_ID":
            return {
                ...state,
                defaultPokemonStates: {...state.defaultPokemonStates, pokemonIndex : action.pokemonIndex}
            };
        case "SET_POKEMON_TRACKER":
            return {
                ...state,
                defaultPokemonStates: {...state.defaultPokemonStates, tracker : action.tracker}
            };
        case "SET_POKEMONS":  
            return {
                ...state,
                defaultPokemonStates: {...state.defaultPokemonStates, pokemons : action.pokemons}
            };
        case "SET_POKEMON_DATA":
            return {
                ...state,
                defaultPokemonStates: {...state.defaultPokemonStates, pokemon : action.pokemon}
            };
        case "SET_POKEMON_SPECIES":
            return {
                ...state,
                defaultPokemonStates: {...state.defaultPokemonStates, speciesData : action.speciesData}
            };
        default:
            return {...state};
    }
};

export default PokemonReducer