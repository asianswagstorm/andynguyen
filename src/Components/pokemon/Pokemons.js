import React, { Component } from "react";
import Search from "./pokemonHelper/Search";
import PokemonCard from "./PokemonCard";
import NoResults from "./pokemonHelper/NoResults";
import "./styles/Pokemons.css";

import axios from "axios";

export default class Pokemons extends Component {
  constructor(props) {
    super(props);
    this.state = {
      by_name: false,
      url: "https://pokeapi.co/api/v2/pokemon/?limit=", //807
      num_of_pokemon: 12,
      pokemon: null,
      isLoading: false,
      searchQuery: "",
      tracker: 0 
    };
  }

  async componentDidMount() {
    const res = await axios.get(
      `${this.state.url}${this.state.num_of_pokemon}`
    ); 
    
    this.setState({ pokemon: res.data["results"] });
    
    document.addEventListener("scroll", this.trackScrolling);
  }

  componentWillUnmount() {
    document.removeEventListener("scroll", this.trackScrolling);
  }

  trackScrolling = () => {
    if (window.innerHeight + window.scrollY >= document.body.offsetHeight) {
      (process.env.NODE_ENV.trim() !== 'production') && console.log("you're at the bottom of the page");
      this.setState({
        isLoading: true,
        error: undefined,
        tracker: this.state.tracker + this.state.num_of_pokemon
      });
      
      setTimeout( () => this.loadMore(), 1000);
    }
  };

  loadMore = () => {

    fetch(`${this.state.url}${this.state.tracker + this.state.num_of_pokemon}`)
      .then(res => res.json())
      .then(
        res => {
          this.setState({
            pokemon: res.results,
            cursor: res.cursor,
            isLoading: false
          });
        },
        error => {
          this.setState({ isLoading: false, error });
        }
      );
    this.setState({ isLoading: false, error: undefined });
  };

  search = e => { //onChange???? 
    e.preventDefault();
    const poke = e.target.elements.pokename.value;
    if (poke !== "") { //clean this up 
      axios
        .get(
          `https://intern-pokedex.myriadapps.com/api/v1/pokemon?name=${poke}`
        )
        .then(res => {
          (process.env.NODE_ENV.trim() !== 'production') && console.log('response',res.data.data);
          if((res.data.data).length === 0 ){
            this.setState({ searchQuery: poke });
          }
          this.setState({ pokemon: res.data.data, by_name: true });
        });
    }
    document.removeEventListener("scroll", this.trackScrolling);
  };

  render() {
    return (
      <div>
        <header className="MyHeader"> 
            <h1 id="my-games">
            <a href="/">
               Pokedex
            </a>
            </h1>
        </header>
        <div className="row">
          <div className="pokemon_search">
            <Search searchResult={this.search} />
          </div>
        </div>
        <div className="pokemon_row">
          {(this.state.pokemon && (this.state.pokemon).length > 0) ? (
            <div className="list_pokemon">
              {this.state.pokemon.map(pokemon => (
                <div className = "single_card_1">
                    <PokemonCard
                    key={pokemon.name}
                    name={pokemon.name}
                    url={pokemon.url}
                    id={pokemon.id}
                    by_name={this.state.by_name}
                    />
                </div>
               
              ))}
              
            </div>
          ) : (
            <NoResults searchQuery = {this.state.searchQuery}/>
          )}
        </div>
      </div>
    );
  }
}
