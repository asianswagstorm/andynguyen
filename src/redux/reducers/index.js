import {combineReducers} from 'redux';
import gamesReducer from './GamesReducer';
import connectFourReducer from './ConnectFourReducer';
import ticTacToeReducer from './TicTacToeReducer';
import PokemonReducer from './PokemonReducer';
import RecipeReducer from './RecipeReducer';
import MusicMasterReducer from './MusicMasterReducer';
const rootReducer = combineReducers({gamesReducer,connectFourReducer,ticTacToeReducer,PokemonReducer,RecipeReducer,MusicMasterReducer});

export default rootReducer;
 