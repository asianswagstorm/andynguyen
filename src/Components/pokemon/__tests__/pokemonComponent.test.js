import React from 'react';
import {shallow} from 'enzyme'; //shallow no dom or children is rendered.
import {Pokemons} from '../Pokemons';
import Search from '../pokemonHelper/Search';
import {defaultPokemonStates} from "../../../redux/reducers/PokemonReducer";

const renderPokemons = (args) => {
    const props = { ...defaultPokemonStates,
                    dispatch: jest.fn(), action_props: {
                    pokemon_action : {
                                        getPokemonsLimit : jest.fn(),
                                        setPokeTracker : jest.fn(),
                                        getPokemonsByName: jest.fn()
                                    }
                    },
                    ...args
    }; 
    return shallow(<Pokemons {...props} />);
};

describe('Pokemons', () => {
    it('should render', () => {
        const wrapper = renderPokemons()
        expect(wrapper).toMatchSnapshot()
    })

    it('should contain pokemon__app', () =>{
        const wrapper = renderPokemons()
        expect(wrapper.find('div.pokemon__app').length).toEqual(1)
    })
})

describe('functions', () => {
    describe('getNumPokemon', () => {
        it('should return 12', () => {
            const wrapper = renderPokemons()
            expect(wrapper.instance().getNumPokemon()).toBe(12);
        })
    })
    describe('SearchPokemon', () => {
        it('should call searchPokemon', () => {
            const wrapper = renderPokemons()
            const spy = jest.spyOn(wrapper.instance(), 'searchPokemon') //Jest mock function. jest.spyOn(object, methodName)
            wrapper.find('div.pokemon__search__button').at(0).simulate('click');            
            expect(spy).toHaveBeenCalled();
        })
        it('should call onChange', () => {
            const wrapper = renderPokemons()
            const onChangeSpy = jest.spyOn(wrapper.instance(), 'change') 
            const searchWrapper = shallow(<Search searchPokemon ={() => jest.spyOn(wrapper.instance(), 'searchPokemon')} onChange={onChangeSpy} searchedPokemon={"aPokemon"} />);
            searchWrapper.find('input').simulate('change', {target: {value: 'pikachu'} });   
            expect(onChangeSpy).toHaveBeenCalled();
        })
    })
})
