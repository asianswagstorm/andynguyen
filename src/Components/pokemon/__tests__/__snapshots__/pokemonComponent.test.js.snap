// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemons should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pokemons
    action_props={
        Object {
            "pokemon_action": Object {
              "getPokemonsByName": [Function],
              "getPokemonsLimit": [Function],
              "setPokeTracker": [Function],
            },
          }
    }
    dispatch={[Function]}
    hasErrors={false}
    pokemon={Object {}}
    pokemonIndex={0}
    pokemons={Array []}
    searchedPokemon=""
    speciesData={Object {}}
    tracker={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Headers
          headerTitle="Pokedex"
          linkTo="#/"
/>,
        <div
          className="row"
>
          <div
                    className="pokemon_search"
          >
                    <div
                              className="pokemon__search__button"
                              onClick={[Function]}
                    >
                              <span
                                        className="icon fa-search"
                                        id="search__icon"
                              />
                    </div>
                    <Search
                              onChange={[Function]}
                              searchPokemon={[Function]}
                              searchedPokemon=""
                    />
          </div>
</div>,
        <div
          className="pokemon_row"
>
          <withRouter(Connect(NoResults))
                    searchedPokemon=""
          />
</div>,
      ],
      "className": "pokemon__app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "headerTitle": "Pokedex",
          "linkTo": "#/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="pokemon_search"
>
            <div
                        className="pokemon__search__button"
                        onClick={[Function]}
            >
                        <span
                                    className="icon fa-search"
                                    id="search__icon"
                        />
            </div>
            <Search
                        onChange={[Function]}
                        searchPokemon={[Function]}
                        searchedPokemon=""
            />
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="pokemon__search__button"
                onClick={[Function]}
>
                <span
                                className="icon fa-search"
                                id="search__icon"
                />
</div>,
              <Search
                onChange={[Function]}
                searchPokemon={[Function]}
                searchedPokemon=""
/>,
            ],
            "className": "pokemon_search",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className="icon fa-search"
                  id="search__icon"
/>,
                "className": "pokemon__search__button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "icon fa-search",
                  "id": "search__icon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "searchPokemon": [Function],
                "searchedPokemon": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <withRouter(Connect(NoResults))
            searchedPokemon=""
/>,
          "className": "pokemon_row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchedPokemon": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Headers
            headerTitle="Pokedex"
            linkTo="#/"
/>,
          <div
            className="row"
>
            <div
                        className="pokemon_search"
            >
                        <div
                                    className="pokemon__search__button"
                                    onClick={[Function]}
                        >
                                    <span
                                                className="icon fa-search"
                                                id="search__icon"
                                    />
                        </div>
                        <Search
                                    onChange={[Function]}
                                    searchPokemon={[Function]}
                                    searchedPokemon=""
                        />
            </div>
</div>,
          <div
            className="pokemon_row"
>
            <withRouter(Connect(NoResults))
                        searchedPokemon=""
            />
</div>,
        ],
        "className": "pokemon__app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "headerTitle": "Pokedex",
            "linkTo": "#/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="pokemon_search"
>
              <div
                            className="pokemon__search__button"
                            onClick={[Function]}
              >
                            <span
                                          className="icon fa-search"
                                          id="search__icon"
                            />
              </div>
              <Search
                            onChange={[Function]}
                            searchPokemon={[Function]}
                            searchedPokemon=""
              />
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="pokemon__search__button"
                  onClick={[Function]}
>
                  <span
                                    className="icon fa-search"
                                    id="search__icon"
                  />
</div>,
                <Search
                  onChange={[Function]}
                  searchPokemon={[Function]}
                  searchedPokemon=""
/>,
              ],
              "className": "pokemon_search",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span
                    className="icon fa-search"
                    id="search__icon"
/>,
                  "className": "pokemon__search__button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "icon fa-search",
                    "id": "search__icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onChange": [Function],
                  "searchPokemon": [Function],
                  "searchedPokemon": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <withRouter(Connect(NoResults))
              searchedPokemon=""
/>,
            "className": "pokemon_row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "searchedPokemon": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifeCycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
